docs/guides-getallsecrets.md

```diff
@@ -33,7 +33,12 @@ Some providers support filtering out a find operation only to a given path, inst
 ### Avoiding name conflicts
 By default, kubernetes Secrets accepts only a given range of characters. `Find` operations will automatically replace any not allowed character with a `_`. So if we have a given secret `a_c` and `a/c` would lead to a naming conflict. 
 
-It is not entirely possible to avoid this behavior, but setting `dataFrom.find.conversionStrategy: Unicode` reduces the collision probability. When using `Unicode`, any invalid character will be replaced by its unicode, in the form of `_UXXXX_`. In this case, the available kubernetes keys would be `a_c` and `a_U2215_c`, hence avoiding most of possible conflicts.
+
+If you happen to have a case where a conflict is happening, you can use the `rewrite` block to apply a regexp on one of the find operations (for more information please refer to [Rewriting Keys from DataFrom](guides-datafrom-rewrite.md)).
+
+You can also set  `dataFrom.find.conversionStrategy: Unicode` to reduce the collistion probability. When using `Unicode`, any invalid character will be replaced by its unicode, in the form of `_UXXXX_`. In this case, the available kubernetes keys would be `a_c` and `a_U2215_c`, hence avoiding most of possible conflicts.
```

Comment:
Is the `conversionStrategy` now obsolete with this feature, should we deprecate it?

Reply:
hmm, I'm not sure!

For me ConversionStrategy plays an important role for users who just uses `dataFrom.find`. If we remove it, `dataFrom.find` for AWS SSM and HashiVault would only work if users explicitly defined a rewrite operation there.  So in this sense, ConversionStrategy ends up being kind of a guardrail.

I would say I am inclined to drop Unicode away, but I'm not sure how the rewrite would work with non-traditional characters (e.g. emojis 😄).

WDYT?

Reply:
I think i'd like to get rid of it and make it explicit with `rewrite`.

> for AWS SSM and HashiVault would only work if users explicitly defined a rewrite operation 

We can return an error with a call to action towards the user.

> considering regexp works fine with weird symbols

I tried that with this PR, unicode can be defined in a yaml manifest and replaces as desired: 

given:

```
鯨-example => foobar
``` 
and
```yaml
- regexp:
    source: ^鯨(.*)
    target: test${1}
```
gives
```
test-example -> foobar
```

Reply:
> We can return an error with a call to action towards the user.

You mean have a validation function before trying to patch the kubernetes secret, right? Otherwise we wouldn't be able to say for sure that the error is indeed the keys not being valid.

Reply:
Yeah, that would work. I just want the user to see: `hey, your keys are invalid, please use rewrite function to fix that <link to docs>`

Reply:
I didn't start testing it yet, but wanted to push so you can take a look. Something in this direction?

I personally don't like that the defaults will be different -> i.e. applied manifests will automatically deprecate

Reply:
yeah, good point with the defaults. We can't change the default behaviour due to our deprecation policy.
We could change the behaviour in a way that allows us to go forward:
1. If `conversionStrategy=Default` and NO rewrite is set: apply the current behaviour (rewrite keys with the Default strat). This is forward and backward compatible
2. If `conversionStrategy=Default` and rewrite is defined: use rewrite INSTEAD OF the `conversionStrategy`. This is forward-compatible and a user needs to opt-in to the rewrite feature in order to change the behaviour. This is backwards-incompatible but only if a user opts into the feature.

Going forward we would then remove the `conversionStrategy` completely and replace it with a `v1beta2` version. WDYT?

Reply:
LGTM! I would just add a third step:
3. If `conversionStrategy=Base64`: generate an event pointing to the deprecation of `conversionStrategy`.

> Going forward we would then remove the conversionStrategy completely and replace it with a v1beta2 version. WDYT?

The idea would be to keep all `v1alpha1`, `v1beta1` and `v1beta2` running, is that correct? I agree it is probably the best approach (although I can only imagine the whole `webhook` again 😭)